parcels/field.py

```diff
@@ -721,6 +721,7 @@ def search_indices_curvilinear(self, x, y, z, xi, yi, ti=-1, time=-1, search2D=F
                          [1, -1, 1, -1]])
         maxIterSearch = 1e6
         it = 0
+        tol = 1.e-10
```

Comment:
from a computational perspective, I'm not a big advocate of tolerance values (as it suggests an algorithm is numerically unstable), but: if that is how it works in this scenario, then it works and that's fine.

Reply:
This `tol` is what we also use in JIT interpolation, see e.g. https://github.com/OceanParcels/parcels/blob/master/parcels/include/index_search.h#L287

Reply:
agreed.